frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -27,6 +28,40 @@ import RunButtonWithTooltip from "../RunButtonWithTooltip";
 
 import StructuredQuery from "metabase-lib/lib/queries/StructuredQuery";
 
+const viewTitleHeaderPropTypes = {
```

Comment:
Prop types of this length make me think:

in cases like this, maybe we can have a separate file named something like `ViewHeader.proptypes.js`, that we import.

Reply:
Doesn't that approach defy the benefit of having proptypes above your functions/classes so that you can better understand the code?

I mean - if we're going to switch to TS at some point, this doesn't matter too much. I'm just curios.

Reply:
Agree with @nemanjaglumac here. TBH I'd keep them, so it motivates us to split stuff when it's too complex and be more conscious about the data flow and design in general :D

Reply:
Yes, if we will split and improve the actual implementation more, also imo it should be even better.